core/adapters/http.go

```diff
@@ -169,54 +170,70 @@ func sendRequest(input models.RunInput, request *http.Request, config HTTPReques
 	}
 	client := &http.Client{Transport: tr}
 
-	response, err := withRetry(client, request, config)
-
+	responseBody, statusCode, err := withRetry(client, request, config)
 	if err != nil {
 		return models.NewRunOutputError(err)
 	}
 
-	defer response.Body.Close()
-
-	source := newMaxBytesReader(response.Body, config.sizeLimit)
-	bytes, err := ioutil.ReadAll(source)
-	if err != nil {
-		return models.NewRunOutputError(err)
-	}
-
-	responseBody := string(bytes)
-	if response.StatusCode >= 400 {
+	// 4xx errors are not retried
+	// We assume this is an error on our side, so retrying will not make any difference
+	if statusCode >= 400 {
```

Comment:
We should not make this change in this hotfix.

Having said that, wouldn't the below make more sense in the future?
```suggestion
	if statusCode < 200 || 300 <= statusCode {
```

Reply:
This code has not changed.

Reply:
Your suggestion seems safer though.

Reply:
Why not make the change now?

Reply:
To be conservative in this hotfix (which we will be rolling out on an accelerated timeline). We can make the change in this thread as part of a normal release.

Reply:
Cool do you wanna add this in another branch off of develop so it doesnt get forgotten?

Reply:
https://www.pivotaltracker.com/story/show/172202268